python/ray/tests/test_client.py

```diff
@@ -135,14 +135,21 @@ def f():
         assert ray.get([]) == []
         assert ray.get([f.remote()]) == ["OK"]
 
+        get_count = 0
+        get_stub = ray.worker.server.GetObject
+
+        def get(req, metadata=None):
+            nonlocal get_count
+            get_count += 1
+            return get_stub(req, metadata=metadata)
+
+        ray.worker.server.GetObject = get
+
         refs = [f.remote() for _ in range(100)]
-        with ray.worker.data_client.lock:
-            req_id_before = ray.worker.data_client._req_id
         assert ray.get(refs) == ["OK" for _ in range(100)]
+
         # Only 1 RPC should be sent.
-        with ray.worker.data_client.lock:
-            assert ray.worker.data_client._req_id == req_id_before + 1, \
-                ray.worker.data_client._req_id
+        assert get_count == 1
```

Comment:
I don't understand this change. Can you please explain?

Reply:
Basically, `ray.get()` is converted to use unary-unary RPC instead of streaming RPC, so we need another way to check if only 1 RPC request is sent. `ray.get()` does not need to serialize with other operations. In future it will be converted to unary-streaming to chunk response data.